Pushwoosh Source doc (#6580)
segmentio/segment-docs: src/connections/sources/catalog/cloud-apps/pushwoosh/index.md

> ---
> title: Pushwoosh Source
> ---

Reviewer: ```suggestion
---
title: Pushwoosh Source
id: MW9K4HgBZz
beta: true
---
```

3. On the next screen, give the Source a name and configure any other settings.

    - The name is used as a label in the Segment app, and Segment creates a related schema name in your warehouse. The name can be anything, but we recommend using something that reflects the source itself and distinguishes among your environments (e.g., Pushwoosh_Prod, Pushwoosh_Staging, Pushwoosh_Dev).

4. Click **Add Source** to save your settings.
5. Copy the Write key from the Segment UI.
6. Log in to your Pushwoosh account and select the Application you want to connect.
7. Navigate to Settings > 3rd party integration > Segment, click the `Configure` button, and enter your write key from your Segment workspace.
8. List the events you want to export using a `,` delimiter in the `Events` field, or leave it blank if you want to export all events, then click `Apply`.

## Stream

Pushwoosh uses our stream Source component to send Segment event data. It uses a server-side track method to send data to Segment. These events are then available in any destination that accepts server-side events, and are available in a schema in your data warehouse.

> Pushwoosh passes the user identifier as the `user_id` property by default.

Reviewer: > Pushwoosh passes the user identifier as the `user_id` property by default.

Could you rewrite this to provide more clarity on the following: 

- is "user identifier" specific to pushwoosh?
- What happens if there is no "user identifier" present, do you send anonymous data with an anonymousId? If so, what is used for anonymousId?

Author: - "is "user identifier" specific to pushwoosh?" - yes, its Pushwoosh User ID (i've fixed it in the doc)
- "What happens if there is no "user identifier" present, do you send anonymous data with an anonymousId? If so, what is used for anonymousId?" - yes, Pushwoosh sent Pushwoosh HWID as the Segment anonymousId (i've noted it in the doc also)

## Events

> The table below lists events that Pushwoosh sends to Segment. These events appear as tables in your warehouse, and as regular events in other Destinations. Pushwoosh includes the `user_id` if available.

Reviewer: > Pushwoosh includes the `user_id` if available.

If referring to the Segment `userId` please change from `user_id` to userId`. What happens when userId is not available, is the event still sent?

Author: Fixed
as i answered in above conversation, If userId not available, Pushwoosh sent Pushwoosh HWID as Segment anonymousId
So, yes, the event still sent

3. On the next screen, give the Source a name and configure any other settings.

    - The name is used as a label in the Segment app, and Segment creates a related schema name in your warehouse. The name can be anything, but Pushwoosh recommends using something that reflects the source itself and distinguishes among your environments (For example: Pushwoosh_Prod, Pushwoosh_Staging, Pushwoosh_Dev).

4. Click **Add Source** to save your settings.
5. Copy the Write key from the Segment UI.
6. Log in to your Pushwoosh account, and select the Application you want to connect.
7. Navigate to **Settings > 3rd party integration > Segment**, click the `Configure` button, and enter your write key from your Segment workspace.
8. List the events you want to export using a `,` delimiter in the `Events` field, or leave it blank if you want to export all events, then click `Apply`.

## Stream

Pushwoosh uses our stream Source component to send Segment event data. It uses a server-side track method to send data to Segment. These events are then available in any destination that accepts server-side events, and are available in a schema in your data warehouse.

> Pushwoosh passes the user identifier as the Segment `userId` property by default, which is equals to Pushwoosh User ID. 
> In other case, if Pushwoosh does not have an associated `userId`, Pushwoosh HWID will be passed as the Segment `anonymousId`.

Reviewer: ```suggestion
By default, Pushwoosh passes the Pushwoosh User ID as the Segment userId property. If there is no associated User ID, Pushwoosh will pass the HWID as the Segment anonymousId.
```
